app/src/lib/stores/git-store.ts

```diff
@@ -138,6 +138,52 @@ export class GitStore {
     return this.emitter.on('did-error', fn)
   }
 
+  /**
+   * Reconcile the local history view with the repository state
+   * after a pull has completed, to include merged remote commits.
+   */
+  public async reconcileHistory(mergeBase: string): Promise<void> {
+    if (this._history.length === 0) {
+      return
+    }
+
+    if (this.requestsInFight.has(LoadingHistoryRequestKey)) {
+      return
+    }
+
+    this.requestsInFight.add(LoadingHistoryRequestKey)
+
+    const commits = await this.performFailableOperation(() =>
+      getCommits(this.repository, `HEAD..${mergeBase}`, CommitBatchSize)
+    )
+    if (!commits) {
```

Comment:
Explicitly check for`null` or `undefined` (can't actually tell what the type is on dotcom).

Reply:
`undefined`, but :thumbsup: anyway